openmdao/recorders/openmdao_server_recorder.py

```diff
@@ -0,0 +1,451 @@
+"""
+Class definition for OpenMDAOServerRecorder, which provides access to the OpenMDAO server endpoints.
+"""
+
+import io
+import requests
+
+import base64
+import json
+import bson
+import numpy as np
+from six import iteritems
+from six.moves import cPickle as pickle
+
+from openmdao.core.driver import Driver
+from openmdao.core.system import System
+from openmdao.recorders.base_recorder import BaseRecorder
+from openmdao.solvers.solver import Solver, NonlinearSolver
+from openmdao.recorders.recording_iteration_stack import \
+    get_formatted_iteration_coordinate, recording_iteration_stack
+
+format_version = 1
+
+class OpenMDAOServerRecorder(BaseRecorder):
+    """
+    Recorder that saves cases to the OpenMDAO server
+
+    Attributes
+    ----------
+    model_viewer_data : dict
+        Dict that holds the data needed to generate N2 diagram.
+    """
+
+    def __init__(self, token, case_name='Case Recording', endpoint='http://www.openmdao.org/visualization', port='', suppress_output = False):
+        """
+        Initialize the OpenMDAOServerRecorder.
+
+        Parameters
+        ----------
+        token: <string>
+            The token to be passed as a passphrase for authentication of each server request
+        case_name: <string>
+            The name this case should be stored under. Default: 'Case Recording'
+        endpoint: <string>
+            The URL (minus port, if not 80) where the server is hosted
+        port: <string>
+            The port which the server is listening on. Default to empty string (port 80)
+        suppress_output: <bool>
+            Indicates if any printing should occur (including printing indicating the visualization URL)
+        """
+        super(OpenMDAOServerRecorder, self).__init__()
+
+        self.model_viewer_data = None
+        self._headers = {'token': token}
+        if port != '':
+            self._endpoint = endpoint + ':' + port + '/case'
+        else:
+            self._endpoint = endpoint + '/case'
+
+        case_data_dict = {
+            'case_name': case_name,
+            'owner': 'temp_owner'
+        }
+
+        case_data = json.dumps(case_data_dict)
+
+        # Post case and get Case ID
+        case_request = requests.post(self._endpoint, data=case_data, headers=self._headers)
+        response = case_request.json()
+        if response['status'] != 'Failed':
+            self._case_id = str(response['case_id'])
+            if not suppress_output:
+                print("Visualization at: http://openmdao.org/visualization/case/" + self._case_id)
```

Comment:
Printing the URL is temporary while I set up a UI for requesting tokens and listing recordings on the web server